test/extensions/filters/http/lua/lua_filter_test.cc

```diff
@@ -1800,6 +1800,39 @@ TEST_F(LuaHttpFilterTest, GetRequestedServerName) {
   EXPECT_EQ(Http::FilterHeadersStatus::Continue, filter_->decodeHeaders(request_headers, true));
 }
 
+// Verify that binary values could also be extracted from dynamicMetadata() in LUA filter.
+TEST_F(LuaHttpFilterTest, GetDynamicMetadataBinaryData) {
+  const std::string SCRIPT{R"EOF(
+    function envoy_on_request(request_handle)
+      local metadata = request_handle:streamInfo():dynamicMetadata():get("envoy.pp")
+      local bin_data = metadata["bin_data"]
+      local str_table = { }
+      local data_length = string.len(metadata["bin_data"])
+      for idx = 1, data_length do
+        if string.byte(bin_data, idx) ~= 0 then
+          str_table[#str_table + 1] = string.char(string.byte(bin_data, idx))
+        end
+      end
+      request_handle:logTrace('Data: ' .. table.concat(str_table, ''))
+    end
+  )EOF"};
+
+  ProtobufWkt::Value metadata_value;
+  constexpr uint8_t buffer[] = {'h', 'e', 0x00, 'l', 'l', 'o'};
+  metadata_value.set_string_value(reinterpret_cast<char const*>(&buffer), sizeof(buffer));
```

Comment:
This looks super weird to me, but apparently uint8_t *p = foo; and uint8_t *p = &foo; both work because static arrays are strange and are actually of type array with pointer decay. I'd defer to others if they have an opinion, but I've always thought of buffer as already basically a pointer, so would expect this to say reinterpret_cast<char const*>(buffer) (lack of &).

Reply:
Changed it to `reinterpret_cast<char const>(buffer)` instead.